include/ginkgo/core/log/stream.hpp

```diff
@@ -164,6 +164,22 @@ class Stream : public Logger {
                                const array<stopping_status>* status,
                                bool stopped) const override;
 
+    [[deprecated(
+        "Please use the version with the additional stopping "
+        "information.")]] void
+    on_iteration_complete(const LinOp* solver, const size_type& num_iterations,
+                          const LinOp* residual, const LinOp* solution,
+                          const LinOp* residual_norm) const override;
+
+    [[deprecated(
+        "Please use the version with the additional stopping "
+        "information.")]] void
+    on_iteration_complete(
+        const LinOp* solver, const size_type& num_iterations,
+        const LinOp* residual, const LinOp* solution,
+        const LinOp* residual_norm,
+        const LinOp* implicit_sq_residual_norm) const override;
+
```

Comment:
are these overloads actually implemented?

Reply:
Yes, see [core/log/stream.cpp](https://github.com/ginkgo-project/ginkgo/pull/1327/files/aa924a68f89bf09c4ec4f3a9912c602f08d99c86#diff-d3fa639a7f8fee5730be0cca4f3bb5b359d95efd5a6b42eac0d1933a2395bc5c)

Reply:
Ah no, I meant record.cpp, papi.cpp etc.

Reply:
They are also there, but somehow not shown in the changes? https://github.com/ginkgo-project/ginkgo/blob/30aea9904fb4d015f42e8a530f2efbea8baeb9ed/core/log/record.cpp#L292